--- libgpg-error-1.32.orig/src/gpgrt-int.h	2018-05-01 12:35:59.000000000 -0500
+++ libgpg-error-1.32/src/gpgrt-int.h	2023-04-11 19:23:21.812801938 -0600
@@ -466,7 +466,7 @@
 
 #if HAVE_W32_SYSTEM
 /* Prototypes for w32-estream.c.  */
-struct cookie_io_functions_s _gpgrt_functions_w32_pollable;
+extern struct cookie_io_functions_s _gpgrt_functions_w32_pollable;
 int _gpgrt_w32_pollable_create (void *_GPGRT__RESTRICT *_GPGRT__RESTRICT cookie,
                                 unsigned int modeflags,
                                 struct cookie_io_functions_s next_functions,

