diff --git a/Win32/DaemonWin32.cpp b/Win32/DaemonWin32.cpp
index 3f9226d1..3eb6c1ee 100644
--- a/Win32/DaemonWin32.cpp
+++ b/Win32/DaemonWin32.cpp
@@ -6,9 +6,9 @@
 #include "Log.h"
 
 #ifdef _WIN32
-#include "Win32/Win32Service.h"
+#include "Win32Service.h"
 #ifdef WIN32_APP
-#include "Win32/Win32App.h"
+#include "Win32App.h"
 #endif
 
 namespace i2p
diff --git a/build/CMakeLists.txt b/build/CMakeLists.txt
index 4d462700..075f9ca6 100644
--- a/build/CMakeLists.txt
+++ b/build/CMakeLists.txt
@@ -35,6 +35,7 @@ set(LIBI2PD_CLIENT_SRC_DIR ../libi2pd_client)
 
 include_directories(${LIBI2PD_SRC_DIR})
 include_directories(${LIBI2PD_CLIENT_SRC_DIR})
+include_directories(../daemon)
 
 set (LIBI2PD_SRC
   "${LIBI2PD_SRC_DIR}/BloomFilter.cpp"
@@ -84,7 +85,7 @@ if (WITH_WEBSOCKETS)
 endif ()
 
 if (WIN32 OR MSYS)
-  list (APPEND LIBI2PD_SRC "${CMAKE_SOURCE_DIR}/I2PEndian.cpp")
+  list (APPEND LIBI2PD_SRC "${LIBI2PD_SRC_DIR}/I2PEndian.cpp")
 endif ()
 
 if (WITH_I2LUA)
@@ -334,7 +335,7 @@ endif()
 
 if (WITH_UPNP)
   find_package ( MiniUPnPc REQUIRED )
-  include_directories( SYSTEM ${MINIUPNPC_INCLUDE_DIR} )
+  include_directories( ${MINIUPNPC_INCLUDE_DIR} )
 endif()
 
 find_package ( ZLIB )
@@ -375,7 +376,7 @@ if (WITH_STATIC AND (MSVC OR MSYS))
 endif ()
 
 # load includes
-include_directories( SYSTEM ${Boost_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR} ${ZLIB_INCLUDE_DIR} )
+include_directories( ${Boost_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR} ${ZLIB_INCLUDE_DIR} )
 
 
 # warn if for meshnet
diff --git a/libi2pd/Identity.cpp b/libi2pd/Identity.cpp
index 23f32bc8..58a2ce27 100644
--- a/libi2pd/Identity.cpp
+++ b/libi2pd/Identity.cpp
@@ -706,7 +706,7 @@ namespace data
 		struct tm tm;
 #ifdef _WIN32
 		gmtime_s(&tm, &t);
-		sprintf_s((char *)(buf + 32), 9, "%04i%02i%02i", tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday);
+		sprintf((char *)(buf + 32), "%04i%02i%02i", tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday);
 #else
 		gmtime_r(&t, &tm);
 		sprintf((char *)(buf + 32), "%04i%02i%02i", tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday);
