diff -u -r boost_1_71_0.orig/boost/beast/http/impl/basic_parser.hpp boost_1_71_0/boost/beast/http/impl/basic_parser.hpp
--- boost_1_71_0.orig/boost/beast/http/impl/basic_parser.hpp
+++ boost_1_71_0/boost/beast/http/impl/basic_parser.hpp
@@ -51,9 +51,9 @@
     }
     // flatten
     net::buffer_copy(net::buffer(
-        buf_.get(), buf_len_), buffers);
+        buf_.get(), size), buffers);
     return put(net::const_buffer{
-        buf_.get(), buf_len_}, ec);
+        buf_.get(), size}, ec);
 }
 
 template<bool isRequest>
